RePublic-Schools/canvas-lms
# Script for exporting weekly quizzes to SR

### Description
Weekly quizzes are going to start to exist, at least in part, in Canvas and we need a way to get them in SR. Canvas will have both the definition of the assessment, and the student answers for each question. This PR adds a script to export quizzes to SR and makes some changes to the SIS import scripts.

When importing users we now also store their SR id (as ext_id) and the SR id has for the school. Both are used when associating an assessment to a particular school and teacher in SR.

To create an assessment in SR we need to send a post to the `/api/v1/assessments/import` endpoint. Right now the url is hard coded to use sandbox, I have another PR coming with the integration of this with PATH that will set the url to SR's prod for when we deploy this. The body of that post gives the assessment's definition and the student answers on the assessment if there are any. The response from SR gives us the id SR now has for that assessment (after saving), nothing uses it yet but it'll be handy to have later.

When generating the assessment's definition, we set the `external_assessment_id` which, according to SR's documentation ([https://republic.schoolrunner.org/api/doc/v1/assessments/import](https://republic.schoolrunner.org/api/doc/v1/assessments/import)), will let us continue to update the assessment that has been created in SR rather than create a new one with each post. We'll use this to update the assessment grades when not all the students take the quiz at the same time, but we may run into problems of it updating the definition of the assessment at some point. For all I know the ideal is that won't happen, but teacher's will be able to change the assessment in SR and they may have a good reason to in the future that I'm not aware of now. Most of the rest of the fields are essentially giving SR the information we have. A complex one is where we have a standard-group associated with a question and we need to give SR the id for the standard from that group that SR has for the school we're syncing the assessment to.

### Test
- Run the migrate script to update your db with the migration script
- Have a blueprint template associated with a course
- Upload some standards to the blueprint template, specifying a particular school id to be used
- Log in to the database and give the teacher of the non-blueprint course (probably your account) the same school id in the pseudonym
- Create a quiz in the blueprint template with at least one question using a standard group that was uploaded and has a standard with the same school id used for the teacher account
- Log in as a student on the non-blueprint course and take the quiz
- I've tested this with SR so I'm confident their API would accept it. Ideally you'd do the same, but it'd be quicker and easier and likely as effective to have you verify that the script compiles the data for SR correctly. So go into the script and comment out the part that actually sends the data to SR and set up a way to see the data that would be sent (either with a breakpoint or logging it out). Along with checking that the data structure sent to SR matches their documentation, check that the standard id sent on the questions is the id of the standard for the school and not the standard group id or anything like that.
